### ISSUE TITLE
dyld: Library not loaded

### ISSUE THREAD

**Issue by minuscorp** [CocoaPods/CocoaPods#3505]
Hi there.

I've been reading all the issues about the error:

```
dyld: Library not loaded: @rpath/Bolts.framework/Bolts
  Referenced from: /private/var/mobile/Containers/Bundle/Application/1542F906-CCE1-4181-AC7C-B5E3EE50E7D7/eBikeMotion.app/eBikeMotion
  Reason: no suitable image found.  Did find:
```

Which makes my application unable to run in a real device (but it runs without any problem in the simulator.
Until certain point I thought that it was an issue with the frameworks I was installing, but after installing manually the original one that was throwing the error, Alamofire, and the next Framework throwing the error was the next one in alphabetical order (Bolts, as you can see in the code snippet)
So I've reached the conclusion that is indeed CocoaPods which is producing these errors. I've got the last version (0.37) with a clean install, Iv'e tried to create a new project, I've tried all the proposed solutions to this issue without any luck, so I have to open an issue, with the hope that someone can help me.

Regards.

**Comment by kylef** [comment]
I have a feeling this may have been fixed via #3489. Would you be able to try running CocoaPods from master and confirm:

```
$ gem install specific_install
$ gem specific_install CocoaPods/CocoaPods
$ pod install
```

**Comment by minuscorp** [comment]
Here's the result of your command:

```
MacBook-Air-de-Gabriel:prueba Gabri$ sudo gem specific_install CocoaPods/CocoaPods
Installing from git@github.com:CocoaPods/CocoaPods.git
Cloning into '/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/d20150506-69456-1527w4y'...
Warning: Permanently added the RSA host key for IP address '192.30.252.128' to the list of known hosts.
Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
ERROR:  While executing gem ... (Errno::ENOENT)
    No such file or directory - /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/d20150506-69456-1527w4y
```

Same result with and without superuser rights.

**Comment by orta** [comment]
^ you might have problems with this command ATM as github is having issues.

**Comment by minuscorp** [comment]
I've done as you told me (your command was giving me problems so I executed ```sudo gem specific_install https://github.com/CocoaPods/CocoaPods.git``` correctly) 
I've done pod install as you told me and this is the error XCode is giving me:

```
dyld: Library not loaded: @rpath/Alamofire.framework/Alamofire
  Referenced from: /private/var/mobile/Containers/Bundle/Application/43BC1C30-F04D-4D8C-8552-390A47DD7711/eBikeMotion.app/eBikeMotion
  Reason: no suitable image found.  Did find:
	/private/var/mobile/Containers/Bundle/Application/43BC1C30-F04D-4D8C-8552-390A47DD7711/eBikeMotion.app/Frameworks/Alamofire.framework/Alamofire: mmap() error 1 at address=0x008B7000, size=0x00054000 segment=__TEXT in Segment::map() mapping /private/var/mobile/Containers/Bundle/Application/43BC1C30-F04D-4D8C-8552-390A47DD7711/eBikeMotion.app/Frameworks/Alamofire.framework/Alamofire
	/private/var/mobile/Containers/Bundle/Application/43BC1C30-F04D-4D8C-8552-390A47DD7711/eBikeMotion.app/Frameworks/Alamofire.framework/Alamofire: mmap() error 1 at address=0x0099B000, size=0x00054000 segment=__TEXT in Segment::map() mapping /private/var/mobile/Containers/Bundle/Application/43BC1C30-F04D-4D8C-8552-390A47DD7711/eBikeMotion.app/Frameworks/Alamofire.framework/Alamofire
	/private/var/mobile/Containers/Bundle/Application/43BC1C30-F04D-4D8C-8552-390A47DD7711/eBikeMotion.app/Frameworks/Alamofire.framework/Alamofire: mmap() error 1 at address=0x00A7F000, size=0x00054000 segment=__TEXT in Segment::map() mapping /private/var/mobile/Containers/Bundle/Application/43BC1C30-F04D-4D8C-8552-390A47DD7711/eBikeMotion.app/Frameworks/Alamofire.framework/Alamofire
(lldb) 
```

Don't pay attention to the name of the framework because if I erase it, the compilation throws the error with the next framework ordered alphabetically.

I've been stuck with this problem for weeks (since 0.36 until last version) and I've tried all fixes around internet (manually creating certs...)

I'm confused that this problem is only thrown in real device execution and NEVER in the simulator.

Regards.

**Comment by kylef** [comment]
Thanks for giving this a try. Could we get a bit more information on this problem so we can attempt to reproduce it. Including your Podfile/Podfile.lock. It would be really helpful if you could build an example project that exhibits the same problem, this allows us to debug the problem far easier without wasting time trying to guess how your project is configured.

**Comment by minuscorp** [comment]
Hi there.

I've created an example project with all the pods I need for the larger project. The project compiles without any problem, runs on simulator, crashes on real device.

Here I leave the project: https://www.dropbox.com/s/2ry9rvdgkuo4z11/test_issue.zip?dl=0

**Comment by gabri15** [comment]
Hi all, i have the same problem. I spent a lot of time searching on internet about the problem, without any good result. I tried on different machines and i have the same error.

I was searching in your project, i use the same frameworks more or less and i have the same error
dyld: Library not loaded,  

When i am trying to deploy the project in a real device.
I hope that the team can investigate this issue!!

Thanks all

**Comment by neonichu** [comment]
Cannot repro this with @minuscorp's project. After changing the team ID to get it to code-sign, this works just fine for me (Xcode 6.3.1, iOS 8.3 device). 

Which Xcode and iOS version combination do you see this on? Are you deploying directly or doing AdHoc?

**Comment by minuscorp** [comment]
Hi there.

Seeing that the problem couldn't been reproduced in other computer made me
think that could be an issue with XCode itself or maybe even the whole
system was wrong.
I formatted and clean installed all the system and now the problem is
gone...

It is a shame that this made me spend all this time (and yours too). The
issue can now be closed.

For future problems like this, I think XCode doesn't handle too well the
signing and cert creation in some ways and it's actually a bug in theirs.

Thank you for all the help.

**Comment by neonichu** [comment]
I guess we discovered a new part of code-signing hell :imp: 

@gabri15 I will close this, as the original issue isn't actionable for us. Please open a new issue with a reproducible case if your problem persists.

**Comment by gabri15** [comment]
@neonichu  the problem was fixed reinstalling Xcode, i don't the reason.
Thanks!!!

**Comment by minuscorp** [comment]
Hi there.

I'm really sorry that I've got to open this issue again...

Yesterday, with the project I provided in the link, it was running well as you checked after a format and a clean install of the system. Afterwards, I started to install the same pods in the main project I'm working on and, sadly, I recieved the same issue again.

I'm really concerned about this, and thinking It was a bad joke, I tried in other Mac. Same issue running in a real device (with the main project)

This time, I'd like to share with you the main project as today is giving the same error in two completely different Mac computers.

This is the link: https://www.dropbox.com/sh/b404pdnvm20mebe/AAD894hKFHgh8JTUeoY7GDFNa?oref=e

Please, tell me that this time at least you get the same error as me and I'm not spending my time in trying to catch a ghost.

Thank you very much.

**Comment by Kueper** [comment]
@minuscorp Were you able to eventually resolve this? I'm having the same issues (and have been for over a month..) A fresh install of Xcode hasn't improved anything, I'm considering formatting my entire computer as you tried... I've also executed all of the other suggestions online and haven't had any success

**Comment by minuscorp** [comment]
@Kueper I solved this issue here: http://stackoverflow.com/questions/30053144/dyld-library-not-loaded-with-cocoapods-0-37-and-xcode-6-3/30166310#30166310

**Comment by Kueper** [comment]
@minuscorp Thanks for the reply! Quick question while I go through your solution - did you also have this issue with a brand new project that had no external pods / frameworks? I get the following on a new project:

>dyld: Library not loaded: @rpath/libswiftCore.dylib
 > Referenced from: /private/var/mobile/Containers/Bundle/Application/A5BC0BD7-0C10-4327-95BA-72323721C676/ProbablyNotGoingToWork.app/ProbablyNotGoingToWork
  >Reason: no suitable image found.  Did find:

For the record, I get the same error as you did with projects that DO have frameworks (i.e. the error occurs for each framework in alphabetical order) but also for libswiftCore...

**Comment by udaysagar1025** [comment]
HI all,

    Which makes my application unable to run in a real device but it runs without any problem in the simulator. This below error getting
        **dyld: Library not loaded: @rpath/APAddressBook.framework/APAddressBook
  Referenced from: /private/var/mobile/Containers/Bundle/Application/8012077F-8606-49E0-811D-3B3D1C3BC031/myapp.app/myapp
  Reason: image not found**

           Please give me any suggestions, how to solve this issue. 

Regards.

**Comment by minuscorp** [comment]
@udaysagar1025 http://stackoverflow.com/questions/30053144/dyld-library-not-loaded-with-cocoapods-0-37-and-xcode-6-3/30166310#30166310 check this out.

**Comment by pedroviana** [comment]
Try to install this https://developer.apple.com/certificationauthority/AppleWWDRCA.cer

**Comment by Santi92** [comment]
Hello, I'm sorry to open this discussion again, I'm very new to IOS and when I went to try out the real device, it came out.

what a pity with my English.

`dyld: Library not loaded: @rpath/Alamofire.framework/Alamofire
  Referenced from: /var/containers/Bundle/Application/B07C286A-FECA-4AF9-89D6-81DA214CB80C/Cabio.app/Cabio
  Reason: no suitable image found.  Did find:`

I've tried everything but nothing
